Yoast/wordpress-seo
# Add builder and watcher for indexable date archives

## Summary

Creates builder and watcher classes for creating and updating date archive indexables.

This PR can be summarized in the following changelog entry:

* N/A

## Test instructions
This PR can be tested by following these steps:

* Make sure you have a presenter for date archives that takes it's value directly from the indexable.
* Go to a date archive, the new presenter should have the same values as the old one.
* Go to your search appearance settings. Change them for the date archive.
* Go to the date archive, the changed settings should be reflected in the presenter output.

## UI changes

## Documentation

## Quality assurance
